node/connection_base.js

```diff
@@ -197,7 +197,11 @@ TChannelConnectionBase.prototype.resetAll = function resetAll(err) {
     outOpKeys.forEach(function eachOutOp(id) {
         var req = self.requests.out[id];
         delete self.requests.out[id];
-        // TODO: shared mutable object... use Object.create(err)?
+        
+        err = errors.TChannelConnectionResetError(err, {
+            connection: self
```

Comment:
you do not want to put non trivial objects on errors. All of these fields are enumerable by default and if someone logs this err it will write a rather large object.

I recommend putting simpler meta data on this object.